### ISSUE TITLE
Enum serialization fails

### ISSUE THREAD

**Issue by scuzzycheese** [zenlist/serde_dynamo#27]
When attempting to serialize and enum variant within a struct and flatten annotation, this occurs:
(I'm running within a Tokio context)

thread 'tokio-runtime-worker' panicked at 'internal error: entered unreachable code', .../src/-280d9361501ba9ca/serde_dynamo-3.0.1/src/ser/serializer_map.rs:31:9

my struct:
```
#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct Message {
    #[serde(with = "ts_milliseconds")]
    pub timestamp: DateTime<Utc>,
    pub identifier: String,
    #[serde(flatten)]
    pub data: Data,
}

#[allow(non_camel_case_types)]
#[derive(Serialize, Deserialize, Debug, Clone)]
pub enum Data {
    string(String),
    binary(Vec<u8>),
}
```

And I can see the serialization code looks incomplete here:

```    
    fn serialize_key<K: ?Sized>(&mut self, _key: &K) -> Result<(), Self::Error>
    where
        K: Serialize,
    {
        unreachable!()
    }

```

Am I missing something?

**Comment by bryanburgers** [comment]
I usually compare `serde_dynamo` output and functionality to `serde_json` output and functionality.

In this case, `serde_json` works great, but `serde_dynamo` panics. So this definitely seems like a bug.

```rust
use chrono::{serde::ts_milliseconds, DateTime, Utc};
use serde::{Deserialize, Serialize};

fn main() {
    let message = Message {
        timestamp: Utc::now(),
        identifier: String::from("id_1"),
        data: Data::string(String::from("the data")),
    };

    match serde_json::to_string_pretty(&message) {
        Ok(json) => println!("{json}"),
        Err(err) => eprintln!("{err}"),
    }
    // => {
    //   "timestamp": 1649104579039,
    //   "identifier": "id_1",
    //   "string": "the data"
    // }

    match serde_dynamo::__aws_sdk_dynamodb_0_9::to_item(&message) {
        Ok(hashmap) => println!("{hashmap:#?}"),
        Err(err) => eprintln!("{err}"),
    }
    // => panic on unreachable!()
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct Message {
    #[serde(with = "ts_milliseconds")]
    pub timestamp: DateTime<Utc>,
    pub identifier: String,
    #[serde(flatten)]
    pub data: Data,
}

#[allow(non_camel_case_types)]
#[derive(Serialize, Deserialize, Debug, Clone)]
pub enum Data {
    string(String),
    binary(Vec<u8>),
}
```

**Comment by scuzzycheese** [comment]
I'm fairly certain it has to do with the Data Enum, or the flatten annotation, because they "string" key was always the place it entered the unreachable! macro if I debugged it.

**Comment by bryanburgers** [comment]
This is fixed in 3.0.2, published just now!

**Comment by scuzzycheese** [comment]
Whoa. That's awesome. Thank you : )
